frontend/src/pages/MapView.tsx

```diff
@@ -18,33 +16,45 @@ const parcelBounds: IParcelListParams = {
   swLong: -123.37
 }
 
-const parcelDetailParams: IParcelDetailParams = {
-  pid: 123
-}
+const parseMapBounds = (bounds: LatLngBounds | null): IParcelListParams | null => {
+  if (!bounds) {
+    return null;
+  }
+  const ne = bounds.getNorthEast();
+  const sw = bounds.getSouthWest();
+  const apiParams = {
+    neLat: ne.lat,
+    neLong: ne.lng,
+    swLat: sw.lat,
+    swLong: sw.lng
+  };
+  return apiParams;
+};
 
 const MapView = () => {
   const parcels = useSelector<RootState, IParcel[]>(state => (state.parcel as IParcelState).parcels);
   const parcelDetail = useSelector<RootState, IParcelDetail | null>(state => (state.parcel as IParcelState).parcelDetail);
-  
   const dispatch = useDispatch();
   useEffect(() => {
     dispatch(fetchParcels(parcelBounds));
-    dispatch(fetchParcelDetail(parcelDetailParams));
-  },[dispatch]);
+  }, [dispatch]);
 
-  return <Container fluid={true}>
-    <Row>
-      <Col>
-        <ParcelList parcels={parcels}></ParcelList>
-      </Col>
-      <Col>
-        <ParcelPopupView parcelDetail={parcelDetail}></ParcelPopupView>
-      </Col>
-      <Col>
-        <Map lat={48.43} lng={-123.37} zoom={14} activeUserId={"0"} />
-      </Col>
-    </Row>
-  </Container>;
+  return (
+    <Map
+      lat={48.43}
+      lng={-123.37}
+      zoom={14}
+      parcels={parcels}
+      activeParcel={parcelDetail}
+      onParcelClick={p => dispatch(fetchParcelDetail({ id: p.id }))}
+      onPopupClose={() => dispatch(storeParcelDetail(null))}
+      onViewportChanged={e => {
```

Comment:
I'm curious, how many requests do we get from this when you move the map around?. I'm wondering if we need to debounce this event to prevent the redux action from being fired a ton of times.

Reply:
Great question. I thought about it but this is a first stab at the requested functionality. Further improvements can come in future PRs